### PR TITLE
Datastore full-text-search column is populated by postgres trigger rather than python 

### LINKED ISSUE #3785: Datastore full-text-search column is populated by postgres trigger rather than python
Datastore tables have a column with the full-text-search indexing. This has hitherto been populated by python code _to_full_text() when loading/updating data. This PR changes it to being generated by a postgres function, triggered by postgres seeing rows being added or updated.

Benefits:
* it works when data gets into the table in ways other than datastore action functions. e.g. columns generated by other postgres triggers. Or ckanext-shift which COPYs data into datastore.
* it's marginally faster (I understand)

### PR DESCRIPTION
**Pull request by davidread**
Fixes #3785 

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

Code is by @wardi, originally part of https://github.com/ckan/ckan/pull/3556

### CODE REVIEW

File: ckanext/datastore/backend/postgres.py [hunk 1]
```diff
@@ -1027,8 +998,11 @@ def upsert_data(context, data_dict):
     fields = _get_fields(context, data_dict)
     field_names = _pluck('id', fields)
     records = data_dict['records']
-    sql_columns = ", ".join(['"%s"' % name.replace(
-        '%', '%%') for name in field_names] + ['"_full_text"'])
+    sql_columns = ", ".join(
+        identifier(name) for name in field_names)
+    sql_full_text = " || ' ' || ".join(
```
Comment (tino097) [inline ckanext/datastore/backend/postgres.py]: this looks like is not used or am i missing something ?

Comment (wardi) [reply ckanext/datastore/backend/postgres.py]: good catch, this was some code from an earlier attempt that I forgot to delete

**Comment by tino097** [comment]
LGTM

**Comment by davidread** [comment]
Thanks @tino097 ! This is 99% your commits @wardi, are you happy for me to merge?

**Comment by wardi** [comment]
@davidread that's fine with me

**Pull request merged**